homekit/StateManager.js

```diff
@@ -360,246 +441,255 @@ module.exports = (device, platform) => {
+				
 				callback()
-			},
-
+			},		
+		
 			TargetHeaterCoolerState: (state, callback) => {
 				const mode = characteristicToMode(state)
-
-				log.easyDebug(device.name + ' -> Setting Target HeaterCooler State:', mode)
+				log.easyDebug(device.name, '(SET) - Target HeaterCooler State:', mode)
 				device.state.mode = mode
 				device.state.active = true
+				
+				updateClimateReact()
+
 				callback()
 			},
-
-			CoolingThresholdTemperature: (temp, callback) => {
-				if (device.usesFahrenheit) {
-					log.easyDebug(device.name + ' -> Setting Cooling Threshold Temperature:', toFahrenheit(temp) + 'ºF')
-				} else {
-					log.easyDebug(device.name + ' -> Setting Cooling Threshold Temperature:', temp + 'ºC')
-				}
+		
+			CoolingThresholdTemperature: (targetTemp, callback) => {
+				if (device.usesFahrenheit)
+					log.easyDebug(device.name, '(SET) - Target Cooling Temperature:', toFahrenheit(targetTemp) + 'ºF')
+				else
+					log.easyDebug(device.name, '(SET) - Target Cooling Temperature:', targetTemp + 'ºC')
 
 				const lastMode = device.HeaterCoolerService.getCharacteristic(Characteristic.TargetHeaterCoolerState).value
 				const mode = characteristicToMode(lastMode)
-
-				log.easyDebug(device.name + ' -> Setting Mode to:', mode)
-				device.state.targetTemperature = temp
+				device.state.targetTemperature = targetTemp								
+				log.easyDebug(device.name, '(SET) - target HeaterCooler State:', mode)
 				device.state.mode = mode
-				device.state.active = true
+
+				updateClimateReact()
+
 				callback()
 			},
-
-			HeatingThresholdTemperature: (temp, callback) => {
-				if (device.usesFahrenheit) {
-					log.easyDebug(device.name + ' -> Setting Heating Threshold Temperature:', toFahrenheit(temp) + 'ºF')
-				} else {
-					log.easyDebug(device.name + ' -> Setting Heating Threshold Temperature:', temp + 'ºC')
-				}
+		
+			HeatingThresholdTemperature: (targetTemp, callback) => {
+				if (device.usesFahrenheit)
+					log.easyDebug(device.name, '(SET) - Target Heating Temperature:', toFahrenheit(targetTemp) + 'ºF')
+				else
+					log.easyDebug(device.name, '(SET) - Target Heating Temperature:', targetTemp + 'ºC')
 
 				const lastMode = device.HeaterCoolerService.getCharacteristic(Characteristic.TargetHeaterCoolerState).value
 				const mode = characteristicToMode(lastMode)
-
-				log.easyDebug(device.name + ' -> Setting Mode to:', mode)
-				device.state.targetTemperature = temp
+				device.state.targetTemperature = targetTemp
+				log.easyDebug(device.name, '(SET) - HeaterCooler State:', mode)
 				device.state.mode = mode
-				device.state.active = true
+
+				updateClimateReact()
+
 				callback()
 			},
 
 			ACSwing: (state, callback) => {
-				const status = state === Characteristic.SwingMode.SWING_ENABLED ? 'SWING_ENABLED' : 'SWING_DISABLED'
+				
+				state = state === Characteristic.SwingMode.SWING_ENABLED ? 'SWING_ENABLED' : 'SWING_DISABLED'
+				log.easyDebug(device.name, '(SET) - AC Swing:', swing)
```

Comment:
Think this should be state as I don't think swing is defined
```suggestion
				log.easyDebug(device.name, '(SET) - AC Swing:', state)
```